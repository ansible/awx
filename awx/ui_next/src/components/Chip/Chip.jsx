import { Chip } from '@patternfly/react-core';
import styled from 'styled-components';

Chip.displayName = 'PFChip';
export default styled(Chip)`
  --pf-c-chip--m-read-only--PaddingTop: 3px;
  --pf-c-chip--m-read-only--PaddingBottom: 3px;
  --pf-c-chip--m-read-only--PaddingLeft: 8px;
  --pf-c-chip--m-read-only--PaddingRight: 8px;

  .pf-c-button {
    --pf-c-button--PaddingTop: 3px;
    --pf-c-button--PaddingBottom: 3px;
    --pf-c-button--PaddingLeft: 8px;
    --pf-c-button--PaddingRight: 8px;
  }
`;
