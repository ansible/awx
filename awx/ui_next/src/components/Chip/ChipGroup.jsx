import { ChipGroup } from '@patternfly/react-core';
import styled from 'styled-components';

export default styled(ChipGroup)`
  --pf-c-chip-group--c-chip--MarginRight: 10px;
  --pf-c-chip-group--c-chip--MarginBottom: 10px;

  > .pf-c-chip.pf-m-overflow .pf-c-button {
    --pf-c-button--PaddingTop: 3px;
    --pf-c-button--PaddingBottom: 3px;
    --pf-c-chip--m-overflow--c-button--PaddingLeft: 8px;
    --pf-c-chip--m-overflow--c-button--PaddingRight: 8px;
  }
`;
