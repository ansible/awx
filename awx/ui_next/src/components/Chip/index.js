export { default as ChipGroup } from './ChipGroup';
export { default as Chip } from './Chip';
export { default as CredentialChip } from './CredentialChip';
