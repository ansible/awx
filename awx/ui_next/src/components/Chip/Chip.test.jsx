import React from 'react';
import { mount } from 'enzyme';
import Chip from './Chip';

describe('Chip', () => {
  test('renders the expected content', () => {
    const wrapper = mount(<Chip />);
    expect(wrapper).toMatchSnapshot();
  });
});
