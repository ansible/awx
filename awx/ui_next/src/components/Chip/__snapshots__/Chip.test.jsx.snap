// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders the expected content 1`] = `
<Chip>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Chip-sc-1rzr8oo-0",
          "isStatic": true,
          "lastClassName": "iczEeI",
          "rules": Array [
            "--pf-c-chip--m-read-only--PaddingTop:3px;--pf-c-chip--m-read-only--PaddingBottom:3px;--pf-c-chip--m-read-only--PaddingLeft:8px;--pf-c-chip--m-read-only--PaddingRight:8px;.pf-c-button{--pf-c-button--PaddingTop:3px;--pf-c-button--PaddingBottom:3px;--pf-c-button--PaddingLeft:8px;--pf-c-button--PaddingRight:8px;}",
          ],
        },
        "displayName": "Chip",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Chip-sc-1rzr8oo-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <PFChip
      className="Chip-sc-1rzr8oo-0 iczEeI"
    >
      <ComponentWithOuia
        component={[Function]}
        componentProps={
          Object {
            "className": "Chip-sc-1rzr8oo-0 iczEeI",
          }
        }
        consumerContext={null}
      >
        <Chip
          className="Chip-sc-1rzr8oo-0 iczEeI"
          closeBtnAriaLabel="close"
          component="div"
          isOverflowChip={false}
          isReadOnly={false}
          onClick={[Function]}
          ouiaContext={
            Object {
              "isOuia": false,
              "ouiaId": null,
            }
          }
          tooltipPosition="top"
        >
          <GenerateId
            prefix="pf-random-id-"
          >
            <div
              className="pf-c-chip Chip-sc-1rzr8oo-0 iczEeI"
            >
              <span
                className="pf-c-chip__text"
                id="pf-random-id-0"
              />
              <ChipButton
                aria-labelledby="remove_pf-random-id-0 pf-random-id-0"
                ariaLabel="close"
                id="remove_pf-random-id-0"
                onClick={[Function]}
              >
                <Component
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-0 pf-random-id-0"
                  className=""
                  id="remove_pf-random-id-0"
                  onClick={[Function]}
                  variant="plain"
                >
                  <ComponentWithOuia
                    component={[Function]}
                    componentProps={
                      Object {
                        "aria-label": "close",
                        "aria-labelledby": "remove_pf-random-id-0 pf-random-id-0",
                        "children": <TimesCircleIcon
                          aria-hidden="true"
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                          title={null}
                        />,
                        "className": "",
                        "id": "remove_pf-random-id-0",
                        "onClick": [Function],
                        "variant": "plain",
                      }
                    }
                    consumerContext={
                      Object {
                        "isOuia": false,
                        "ouiaId": null,
                      }
                    }
                  >
                    <Button
                      aria-label="close"
                      aria-labelledby="remove_pf-random-id-0 pf-random-id-0"
                      className=""
                      id="remove_pf-random-id-0"
                      onClick={[Function]}
                      ouiaContext={
                        Object {
                          "isOuia": false,
                          "ouiaId": null,
                        }
                      }
                      variant="plain"
                    >
                      <button
                        aria-disabled={null}
                        aria-label="close"
                        aria-labelledby="remove_pf-random-id-0 pf-random-id-0"
                        className="pf-c-button pf-m-plain"
                        disabled={false}
                        id="remove_pf-random-id-0"
                        onClick={[Function]}
                        tabIndex={null}
                        type="button"
                      >
                        <TimesCircleIcon
                          aria-hidden="true"
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                          title={null}
                        >
                          <svg
                            aria-hidden="true"
                            aria-labelledby={null}
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                              transform=""
                            />
                          </svg>
                        </TimesCircleIcon>
                      </button>
                    </Button>
                  </ComponentWithOuia>
                </Component>
              </ChipButton>
            </div>
          </GenerateId>
        </Chip>
      </ComponentWithOuia>
    </PFChip>
  </StyledComponent>
</Chip>
`;
